src/main/io/gps.c

```diff
@@ -1805,22 +2352,19 @@ static bool gpsNewFrameUBLOX(uint8_t data)
             _step++;
             _ck_b += (_ck_a += data);       // checksum byte
             _payload_length += (uint16_t)(data << 8);
-#if DEBUG_UBLOX_FRAMES
-    debug[3] = _payload_length;
-#endif
-            if (_payload_length > UBLOX_PAYLOAD_SIZE) {
-                _skip_packet = true;
-            }
+
             _payload_counter = 0;   // prepare to receive payload
             if (_payload_length == 0) {
                 _step = 7;
             }
+            _buffer.bytes[_payload_counter] = data;
+            if (_payload_length > UBLOX_PAYLOAD_SIZE) {
+                _payload_length = UBLOX_PAYLOAD_SIZE;
+            }
             break;
         case 6:
             _ck_b += (_ck_a += data);       // checksum byte
-            if (_payload_counter < UBLOX_PAYLOAD_SIZE) {
-                _buffer.bytes[_payload_counter] = data;
-            }
+            _buffer.bytes[_payload_counter] = data;
             if (++_payload_counter >= _payload_length) {
                 _step++;
             }
```

Comment:
following logic will `gpsData.errors++;` twice for most wrong checksum (both ck bytes mismathch). It will count packets with _ck_a fail and ck_b ok (0.5% of failed packets)

Reply:
At present we don't log gpsData.errors but I can see that would be a useful thing to do.
It would be best to count one error only once :-)

Reply:
Reverted [42fc8c04f](https://github.com/betaflight/betaflight/pull/12799/commits/42fc8c04fdbc436c9ef196b88f0764ffcbb9239f)